src/main/java/org/cyclonedx/gradle/SbomGraphProvider.java

```diff
@@ -0,0 +1,150 @@
+        }
+
+        project.getLogger().info(MESSAGE_RESOLVING_DEPS);
+
+        final Map<SbomComponentId, SbomComponent> graph = Stream.concat(
+                        Stream.of(project), project.getSubprojects().stream())
+                .filter(project -> !shouldSkipProject(project))
+                .flatMap(this::traverseProject)
+                .reduce(new HashMap<>(), DependencyUtils::mergeGraphs);
+
+        return buildSbomGraph(graph);
+    }
+
+    private SbomGraph buildSbomGraph(final Map<SbomComponentId, SbomComponent> graph) {
+
+        final Optional<SbomComponent> rootProject = DependencyUtils.findRootComponent(project, graph);
+        if (rootProject.isPresent()) {
+            DependencyUtils.connectRootWithSubProjects(
+                    project, rootProject.get().getId(), graph);
+            return new SbomGraph(graph, rootProject.get());
+        } else {
+            project.getLogger().debug("CycloneDX: root project not found. Constructing it.");
+            final SbomComponentId rootProjectId = new SbomComponentId(
+                    project.getGroup().toString(),
+                    project.getName(),
+                    project.getVersion().toString(),
+                    "");
+            final SbomComponent sbomComponent = new SbomComponent.Builder()
+                    .withId(rootProjectId)
+                    .withDependencyComponents(new HashSet<>())
+                    .withInScopeConfigurations(new HashSet<>())
+                    .withLicenses(new ArrayList<>())
+                    .build();
+
+            return new SbomGraph(graph, sbomComponent);
+        }
+    }
+
+    private Stream<Map<SbomComponentId, SbomComponent>> traverseProject(final Project project) {
+
+        final DependencyGraphTraverser traverser =
+                new DependencyGraphTraverser(project.getLogger(), getArtifacts(), mavenLookup, task);
+
+        return project.getConfigurations().stream()
+                .filter(configuration -> shouldIncludeConfiguration(configuration)
+                        && !shouldSkipConfiguration(configuration)
+                        && configuration.isCanBeResolved())
+                .map(config -> traverser.traverseGraph(
+                        config.getIncoming().getResolutionResult().getRoot(), project.getName(), config.getName()));
+    }
+
+    private Map<ComponentIdentifier, File> getArtifacts() {
+        return Stream.concat(Stream.of(project), project.getSubprojects().stream())
+                .filter(project -> !shouldSkipProject(project))
+                .flatMap(project -> project.getConfigurations().stream())
+                .filter(configuration -> shouldIncludeConfiguration(configuration)
+                        && !shouldSkipConfiguration(configuration)
+                        && configuration.isCanBeResolved())
+                .flatMap(config -> config.getIncoming().getArtifacts().getArtifacts().stream())
```

Comment:
Add lenient `artifactView` to avoid errors when selecting variants in Android projects:

```suggestion
                .flatMap(config -> config.getIncoming().artifactView(view -> {view.lenient(true);}).getArtifacts().getArtifacts().stream())
```

Note that this suggestion will likely fail `spotless` and this should be run through `./gradlew :spotlessApply`.